Update package for TileDB 2.29

ihnorton opened the pull request.

ihnorton committed:
Bump version target to libtiledb 2.29

jdblischak reviewed (commented):
If this is for deployment in the legacy cloud saas, then we need a "for-cloud" variant submitted to the `for-cloud` branch. A binary built from main won't be installable in our cloud environment

xref: https://github.com/conda-forge/libtiledb-sql-feedstock/issues/191, https://github.com/TileDB-Inc/libtiledb-sql-feedstock/pull/3

ihnorton committed:
MNT: Re-rendered with conda-smithy 3.52.3 and conda-forge-pinning 2025.10.29.09.00.09

ihnorton commented:
@jdblischak thanks, updated and re-targeted. I also:
- changed the default branch to `for-cloud`
- added a README note [here](https://github.com/TileDB-Inc/libtiledb-sql-feedstock/commit/89f26c81f593170536260e32535063f0c1a9f51b)

jdblischak reviewed (changes requested):
Should bump the version to [0.36.1](https://github.com/TileDB-Inc/TileDB-MariaDB/releases/tag/0.36.1) or the build number to 1

jdblischak commented:
> * changed the default branch to `for-cloud`

👍 

> * added a README note [here](https://github.com/TileDB-Inc/libtiledb-sql-feedstock/commit/89f26c81f593170536260e32535063f0c1a9f51b)

That will be useful for the next time we need to update, but unfortunately it will be overwritten the next time the feedstock is rerendered

Shelnutt2 commented:
Should we cut a new release so we have a tag like 0.37.0 with 2.29.1 in both the docker images and the feedstock? (Not we haven't updated the TIleDB-MariaDB repo yet for 2.29.1)

ihnorton commented:
https://github.com/TileDB-Inc/TileDB-MariaDB/pull/397

ihnorton committed:
Update to TileDB-MariaDB 0.37.0

ihnorton committed:
sha256
